### PR TITLE
Reinstate CubeList._repr_html_()

### PR DESCRIPTION
**Pull request by pp-mo**
## 🚀 Pull Request

To address #4973 

**NOTE:** aiming this at a bugfix release, so it is derived from, and targetting '3.3.x' branch.

**TODO**:
  - [x] add whatsnew
      - ( create appropriate new whatsnew/latest.rst, on v3.3.x aiming  for 3.3.1 )

### CODE REVIEW

**Comment by pp-mo** [comment]
Pure tests to start with -- proving that the Cubelist._repr_html_ test **_will fail_**.

**Comment by pp-mo** [comment]
... and that should fix it.

**Comment by pp-mo** [comment]
Ok, naming @trexfeathers and @lbdreyer as possible reviewers.
@lbdreyer simply because I'm still easing into Pytest usage.

We need to consider the implications for this going in a 3.3.1 release.
**TODO**: (for me) work out what a whatsnew "latest.rst" should look like in this case + initiate it.

**Comment by lbdreyer** [comment]
Regarding

> TODO: (for me) work out what a whatsnew "latest.rst" should look like in this case + initiate it.

Rather than editing latest.rst. I think you just need to edit the 3.3.rst to add a section for bug fixes (as we did in https://github.com/SciTools/iris/pull/4604/files) and then add your whats new entry to this?

**Comment by pp-mo** [comment]
> Regarding
> 
> > TODO: (for me) work out what a whatsnew "latest.rst" should look like in this case + initiate it.
> 
> Rather than editing latest.rst. I think you just need to edit the 3.3.rst to add a section for bug fixes (as we did in https://github.com/SciTools/iris/pull/4604/files) and then add your whats new entry to this?

Ok thanks, that makes sense !
I tried to consult the developer docs, but it isn't really clear about this IMHO

**Review by lbdreyer — CHANGES_REQUESTED** [review]
Tests look great! Just a few small comments

File: lib/iris/tests/unit/cube/test_Cube.py [hunk 1]
```diff
@@ -2876,64 +2876,176 @@ def test_cell_method_correct_order(self):
...
+        res = simplecube._dimensional_metadata(
+            simplecube.ancillary_variable("quality_flag")
         )
-        assert res == cube.ancillary_variable("quality_flag")
+        assert res == simplecube.ancillary_variable("quality_flag")
 
-    def test_two_with_same_name(self, cube):
-        # If a cube has two _DimensionalMetadata objects with the same name, the
+    def test_two_with_same_name(self, simplecube):
+        # If a simplecube has two _DimensionalMetadata objects with the same name, the
```
[the remark this answers is not in the document]
Comment (pp-mo) [reply lib/iris/tests/unit/cube/test_Cube.py:2932]: 👍 Rename+replace gone too far !

File: lib/iris/tests/unit/cube/test_Cube.py [hunk 2]
```diff
@@ -2876,64 +2876,176 @@ def test_cell_method_correct_order(self):
...
+        res = simplecube._dimensional_metadata(
+            simplecube.cell_measure("wibble")
+        )
+        assert res == simplecube.cell_measure("wibble")
+
+
+class TestReprs:
+    """
+    Confirm that str(cube), repr(cube) and cube.summary() work by creating a fresh
+    fresh :class:`iris._representation.cube_printout.CubePrinter` object, and using it
```
[the remark this answers is not in the document]
Comment (pp-mo) [reply lib/iris/tests/unit/cube/test_Cube.py:2953]: :+1: !

File: lib/iris/tests/unit/cube/test_CubeList.py [hunk 3]
```diff
@@ -735,5 +735,39 @@ def test_copy(self):
...
+    `iris.tests.unit.experimental.representation.test_CubeListRepresentation`
+    """
+
+    @staticmethod
+    def test__repr_html_():
+        test_cubelist = CubeList([])
+
+        target = "iris.experimental.representation.CubeListRepresentation"
+        instance_mock = mock.MagicMock(
+            repr_html=mock.MagicMock()  # return_value='')  # NB this must return a string
```
[the remark this answers is not in the document]
Comment (pp-mo) [reply lib/iris/tests/unit/cube/test_CubeList.py:756]: :+1: good spot !
Copied from the 'cube' case, where it did matter -- will fix.

**Comment by pp-mo** [comment]
Thanks @lbdreyer 
Hope I have fixed those things now.

N.B. I am just about to check actual notebook output, but I tidied my conda + it is taking ages to build a test env ...
Best to wait for that.

**Comment by pp-mo** [comment]
Ok I think this will work now : 
Fixed the glitch in the whatsnew.
I also checked that it does work as expected in a notebook :heavy_check_mark:

File: docs/src/whatsnew/3.3.rst [hunk 4]
```diff
@@ -31,6 +31,21 @@ This document explains the changes made to Iris for this release
    any issues or feature requests for improving Iris. Enjoy!
 
 
+v3.3.1 |build_date| [unreleased]
+================================
+
+.. dropdown:: :opticon:`alert` v3.X.X Patches
```
Comment (lbdreyer) [inline docs/src/whatsnew/3.3.rst:37]:
```suggestion
.. dropdown:: :opticon:`alert` v3.3.1 Patches
```

This will also need updating

**Review by lbdreyer — APPROVED** [review]

**Comment by lbdreyer** [comment]
Whoops, Looks like there was a problem with the docs build from this merge: https://readthedocs.org/projects/scitools-iris/builds/18152663/

**Comment by pp-mo** [comment]
> Whoops, Looks like there was a problem with the docs build from this merge: https://readthedocs.org/projects/scitools-iris/builds/18152663/

Some kind of HDF error, could be a bit of a random glitch ?
Just trying a re-build : https://readthedocs.org/projects/scitools-iris/builds/18168111/

**Comment by pp-mo** [comment]
Ok it has built, but my new whatsnew is not properly contained in the dropdown section
Need more indent.
However, I think we should shortly be adding to this section in #4933, so I will fix it there ...

**Pull request merged**